clients/git/client.go

```diff
@@ -207,3 +261,84 @@ func (c *Client) Close() error {
 	}
 	return nil
 }
+
+func (c *Client) GetBranch(branch string) (*clients.BranchRef, error) {
```

Comment:
Similarly, if this is primarily for branch protection, I'd be curious if `clients.ErrUnsupportedFeature` is relevant. 

But fine with this as written, again just looking for possible discussion points.

Reply:
Included a TODO.

Reply:
I can see both ways. Extending BranchRef to include additional details could be helpful in the future. For now, I am going to leave it as it is.